services/insights-handler/internal/handler/main.go

```diff
@@ -0,0 +1,749 @@
+			return apiErr
+		}
+
+		apiErr = h.pushFactsToLagoonApi(facts, resource)
+		if apiErr != nil {
+			return apiErr
+		}
+	}
+
+	return nil
+}
+
+func (h *Messaging) deleteExistingFactsBySource(apiClient graphql.Client, environment lagoonclient.Environment, source string, project lagoonclient.Project) error {
+	// Remove existing facts from source first
+	_, err := lagoonclient.DeleteFactsFromSource(context.TODO(), apiClient, environment.Id, source)
+	if err != nil {
+		return err
+	}
+
+	log.Println("--------------------")
+	log.Printf("Previous facts deleted for '%s:%s' and source '%s'", project.Name, environment.Name, source)
+	log.Println("--------------------")
+	return nil
+}
+
+func (h *Messaging) getApiClient() graphql.Client {
+	apiClient := graphql.NewClient(h.LagoonAPI.Endpoint, &http.Client{Transport: &authedTransport{wrapped: http.DefaultTransport, h: h}})
+	return apiClient
+}
+
+func determineResourceFromLagoonAPI(apiClient graphql.Client, resource ResourceDestination) (lagoonclient.Project, lagoonclient.Environment, error) {
+	// Get project data (we need the project ID to be able to utilise the environmentByName query)
+	project, err := lagoonclient.GetProjectByName(context.TODO(), apiClient, resource.Project)
+	if err != nil {
+		return lagoonclient.Project{}, lagoonclient.Environment{}, err
+	}
+
+	if project.Id == 0 || project.Name == "" {
+		return lagoonclient.Project{}, lagoonclient.Environment{}, fmt.Errorf("error: unable to determine resource destination (does %s:%s exist?)", resource.Project, resource.Environment)
+	}
+
+	environment, err := lagoonclient.GetEnvironmentFromName(context.TODO(), apiClient, resource.Environment, project.Id)
+	if err != nil {
+		return lagoonclient.Project{}, lagoonclient.Environment{}, err
+	}
+	return project, environment, nil
+}
+
+func (h *Messaging) sendToLagoonS3(incoming *InsightsMessage, insights InsightsData, resource ResourceDestination) (err error) {
+	// strip http/s protocol from origin
+	u, _ := url.Parse(h.S3Config.S3Origin)
+	if u.Scheme == "http" || u.Scheme == "https" {
+		h.S3Config.S3Origin = u.Host
+	}
+
+	// Push to s3 bucket
+	minioClient, err := minio.New(h.S3Config.S3Origin, &minio.Options{
+		Creds:  credentials.NewStaticV4(h.S3Config.AccessKeyId, h.S3Config.SecretAccessKey, ""),
+		Secure: h.S3Config.UseSSL,
+	})
+	if err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+	err = minioClient.MakeBucket(ctx, h.S3Config.Bucket, minio.MakeBucketOptions{Region: h.S3Config.Region})
+	if err != nil {
+		exists, errBucketExists := minioClient.BucketExists(ctx, h.S3Config.Bucket)
+		if errBucketExists != nil && !exists {
+			return err
+		}
+	} else {
+		log.Printf("Successfully created %s\n", h.S3Config.Bucket)
+	}
+
+	if incoming.Payload != nil {
+		b, err := json.Marshal(incoming)
+		if err != nil {
+			return err
+		}
+
+		objectName := strings.ToLower(fmt.Sprintf("%s-%s-%s-%s.json", insights.InputType, resource.Project, resource.Environment, resource.Service))
```

Comment:
I think this can be changed to `ci-python-control-k8s-python37-python-sbom.json` (and likeewise the gz, but I don't think we need to keep that one)

![image](https://user-images.githubusercontent.com/4373945/156246836-77102992-4ed5-4ff1-ba7c-d34b5b4334b0.png)